App/dfu/viewmodel/DfuViewModel.swift

```diff
@@ -85,9 +85,11 @@ class DfuViewModel : ObservableObject, DFUProgressDelegate, DFUServiceDelegate {
     func onFileSelected(selected file: ZipFile) throws {
         _ = try DFUFirmware(
             urlToZipFile: file.url,
-            type: DFUFirmwareType.softdeviceBootloaderApplication
+            type: .softdeviceBootloaderApplication
         )
-        zipFile = file
+        DispatchQueue.main.async { [weak self] in
```

Comment:
Maybe it will be better to keep threading operations in one place (where we download a file) and keep this function simple.

Reply:
I agree.

Reply:
Fixed